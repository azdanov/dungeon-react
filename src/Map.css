.app {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.row {
  color: rgba(104, 159, 56, 1);
  display: flex;
  justify-content: center;
}

.flex-container {
  border: 1px solid rgba(0, 0, 0, 0.2);
}

.cell {
  text-align: center;
  height: 1.5rem;
  width: 1.5rem;
  outline: 1px solid rgba(97, 97, 97, 1);
  background: rgba(97, 97, 97, 0.4);
}

.wall {
  outline: none;
  /* body has same color as walls */
  background: transparent;
}

.room {
  outline: 1px solid rgba(97, 97, 97, 0.55);
  background: rgba(189, 195, 199, 1);
}

.bridge {
  background: rgba(117, 117, 117, 1);
}
