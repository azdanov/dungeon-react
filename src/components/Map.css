.row {
  color: rgba(104, 159, 56, 1);
  display: flex;
  justify-content: center;
}

.flex-container {
  border: 1px solid rgba(0, 0, 0, 0.2);
}
