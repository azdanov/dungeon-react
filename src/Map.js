// @flow
/* eslint-disable react/no-array-index-key */
import React from 'react';

import './Map.css';
import dungeon from './utils/DungeonMaster';

export function getClassName(cell?: string | number) {
  const className = 'cell';
  switch (cell) {
    case 0: {
      return `${className} bridge`;
    }
    case 1: {
      return `${className} wall`;
    }
    case 'P': {
      return `${className} player`;
    }
    case 'E': {
      return `${className} enemy`;
    }
    case 'H': {
      return `${className} health`;
    }
    case 'S': {
      return `${className} strength`;
    }
    default: {
      return `${className} room`;
    }
  }
}

const Map = () => (
  <div className="app">
    {dungeon.map((row, rowIndex) => (
      <div className="row" key={rowIndex}>
        {row.map((cell, cellIndex) => {
          const className = getClassName(cell);
          return (
            <div className={className} key={`${rowIndex}${cellIndex}`}>
              {cell}
            </div>
          );
        })}
      </div>
    ))}
  </div>
);

export default Map;
