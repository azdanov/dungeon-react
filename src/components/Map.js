// @flow
/* eslint-disable react/no-array-index-key */
import React from 'react';
import type { cell } from '../dungeon/DungeonKeeper';

import Cell from './Cell';
import './Map.css';

const Map = (props: { dungeonMap: Array<Array<cell>> }) => (
  <div>
    {props.dungeonMap.map((row, rowIndex) => (
      <div className="row" key={rowIndex}>
        {row.map((cellObj, cellIndex) => (
          <Cell
            key={cellIndex}
            type={cellObj.type}
            symbol={cellObj.symbol}
            visible={cellObj.visible}
            visited={cellObj.visited}
          />
        ))}
      </div>
    ))}
  </div>
);

export default Map;
