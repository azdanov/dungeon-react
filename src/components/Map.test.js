import React from 'react';
import ReactDOM from 'react-dom';
import Map from './Map';

describe('Map', () => {
  xit('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Map />, div);
  });
});
