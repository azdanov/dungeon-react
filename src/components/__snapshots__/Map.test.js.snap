// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders without crashing 1`] = `<div />`;
